cmake/checks/check_02_compiler_features.cmake

```diff
@@ -501,3 +501,44 @@ IF(NOT CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
 
   RESET_CMAKE_REQUIRED()
 ENDIF()
+
+
+#
+# CCache support is only really useful for developers. 
+# We therefore disable the feature by default.
+#
+# See https://invent.kde.org/kde/konsole/-/merge_requests/26/diffs
+# and https://stackoverflow.com/a/24305849
+#
+SET(BUILD_USING_CCACHE FALSE CACHE BOOL "Enable ccache support")
+
+MESSAGE(STATUS "")
+MESSAGE(STATUS "Looking for CCache.")
+IF(BUILD_USING_CCACHE)
+  #
+  # Built-in CCache support requires CMake version 2.8 or newer
+  #
+  IF(CMAKE_VERSION VERSION_LESS 2.8)
```

Comment:
We already require 2.8.12 in the top level directory, so this is not necessary

Reply:
Please remove the if conditional on the cmake version as @bangerth pointed out